src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+    private static final Status.StatusName DEFAULT_STATUS_NAME
+            = Status.StatusName.PENDING;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final OrderStatusRepository statusRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto addOrder(Long userId, OrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartService.getNonEmptyCartByUserId(userId);
+        Order newOrder = createOrder(userId, shoppingCart, requestDto);
+        orderRepository.save(newOrder);
+        shoppingCartService.clearShoppingCart(shoppingCart.getId());
+        return orderMapper.toDto(newOrder);
+    }
+
+    @Override
+    public Set<OrderResponseDto> getAllOrders(Long userId) {
+        Set<OrderResponseDto> set = new TreeSet<>();
+        orderRepository.findAllByUserId(userId)
+                .forEach(order -> set.add(orderMapper.toDto(order)));
+        return set;
+    }
+
+    @Override
+    public Set<OrderItemResponseDto> getItemsByOrderId(Long orderId, Long userId) {
+        Order order = getByIdAndUserId(orderId, userId);
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderItemResponseDto getItemByOrderAndItemId(Long itemId, Long orderId, Long userId) {
+        return orderItemMapper.toDto(getByIdAndUserId(orderId, userId).getOrderItems().stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findAny().orElseThrow(()
+                        -> new EntityNotFoundException("There is no item by these "
+                        + "item id and order id: " + itemId + " and " + orderId)));
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(StatusDto statusDto, Long orderId) {
+        Status status = getByName(statusDto.getName());
+        Order order = orderRepository.findById(orderId).orElseThrow(()
+                -> new EntityNotFoundException("No order by id " + orderId));
+        order.setStatus(status);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    private Set<OrderItem> mapCartItemToOrderItem(Set<CartItem> cartItems, Order order) {
+        Set<OrderItem> set = new HashSet<>();
+        cartItems.stream()
+                .map(item -> createOrderItem(item, order))
+                .forEach(set::add);
+        return set;
+    }
+
+    private BigDecimal getTotal(Set<CartItem> items) {
+        return items.stream()
+                .map(item -> item.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(item.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private Status getByName(Status.StatusName name) {

```

Comment:
```suggestion
    private Status getOrederStatusByName(Status.StatusName name) {
```
or move to it's own service and leave the name as it is